src/modules/Game.class.js

```diff
@@ -1,68 +1,301 @@
+    this.score = 0;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.state;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  start() {
+    this.status = Game.STATUS.PLAYING;
+    this.state = this.initialState.map((row) => [...row]);
+    this.addNewNumber();
+    this.addNewNumber();
+  }
+
+  restart() {
+    this.status = Game.STATUS.IDLE;
+    this.state = this.initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  addNewNumber() {
+    const emptyPositions = this.getEmptyPosition();
+
+    if (emptyPositions.length === 0) {
+      return;
+    }
+
+    const randomIndex = Math.floor(Math.random() * emptyPositions.length);
+    const [row, col] = emptyPositions[randomIndex];
+
+    this.state[row][col] = Math.random() < 0.9 ? 2 : 4;
+  }
+
+  getEmptyPosition() {
+    const emptyPositions = [];
+
+    for (let row = 0; row < 4; row++) {
+      for (let col = 0; col < 4; col++) {
+        if (this.state[row][col] === 0) {
+          emptyPositions.push([row, col]);
+        }
+      }
+    }
+
+    return emptyPositions;
+  }
+
+  moveLeft() {
+    if (this.getStatus() === Game.STATUS.PLAYING) {
+      const size = this.state.length;
+      const state = this.getState();
+      let moved = false;
+
+      for (let row = 0; row < size; row++) {
+        const newRow = [];
+        const merged = Array(size).fill(false);
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== 0) {
+            newRow.push(state[row][col]);
+          }
+        }
+
+        for (let i = 0; i < newRow.length - 1; i++) {
+          if (newRow[i] === newRow[i + 1] && !merged[i] && !merged[i + 1]) {
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow.splice(i + 1, 1);
+            merged[i] = true;
+          }
+        }
+
+        while (newRow.length < size) {
+          newRow.push(0);
+        }
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== newRow[col]) {
+            state[row][col] = newRow[col];
+            moved = true;
+          }
+        }
+      }
+
+      this.handleMove(moved);
+    }
+  }
+
+  moveRight() {
+    if (this.getStatus() === Game.STATUS.PLAYING) {
+      const size = this.state.length;
+      const state = this.getState();
+      let moved = false;
+
+      for (let row = 0; row < size; row++) {
+        const newRow = [];
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== 0) {
+            newRow.push(state[row][col]);
+          }
+        }
+
+        const merged = Array(newRow.length).fill(false);
+
+        for (let i = newRow.length - 1; i > 0; i--) {
+          if (newRow[i] === newRow[i - 1] && !merged[i] && !merged[i - 1]) {
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow.splice(i - 1, 1);
+            merged.splice(i - 1, 1);
+            merged[i - 1] = true;
```

Comment:
The merging logic in the 'moveRight' method is similar to that in 'moveLeft'. To reduce code duplication, consider creating a method that handles the merging of tiles, which can be reused in both 'moveLeft' and 'moveRight'.